Latest requirements

jburel opened the pull request:
* Update the version to 5.3
 * Use the latest requirements file to match the rest of the doc

jburel committed:
Bump version

jburel committed:
Use the new requirements file to match rest of the doc

jburel committed:
Point to latest

joshmoore commented on .travis.yml:
```diff
@@ -16,7 +16,7 @@ before_install:
 script:
     - cd linux/test
-    - ./docker-build.sh $ENV
+    - OMEROVER=OMERO-DEV-latest ./docker-build.sh $ENV
```
The idea is to drop this once 5.3.0 is released?

jburel replied:
only to have travis to pass otherwise m9 will be used with contain omero-marshal dependencies installed using git

joshmoore commented on linux/step05_centos6_py27_ius_nginx.sh:
```diff
@@ -30,11 +30,16 @@ set -u
 	#start-latest
 	/home/omero/omeroenv/bin/pip2.7 install -r $file
 	#end-latest
+elif $(is_less_than $OMEROVER 5.3); then
```
:+1: for trying to keep backwards compatibility but I guess there's a question of whether or not we try to do so. Are we basically saying `master` is now 5.3 and therefore we either branch off `dev_5_2` or tell anyone who wants to to use a tag?

jburel replied:
the various switches in place are mainly  there if someone needs to "build" a 5.1, etc. server
happy to remove them since previously we did not have branches for that repo only develop

jburel replied:
happy to remove the various versions check since we now have dev_5_1 and dev_5_2 branches
this will also simplify the code and remove the need of the version in utils

joshmoore reviewed (commented):
Few questions, largely clarifications of the "major-version-bump" process for this repo.

cc: @sbesson

jburel committed:
Remove version check

Only install 5.3 requirements

jburel committed:
Install latest from download

jburel committed:
Point to latest

jburel committed:
Roll back travis change

jburel committed:
Add dependencies

jburel committed:
Remove usage of version

jburel committed:
Remove utils. no longer needed

jburel committed:
Remove install from requirements

The file is now empty

jburel committed:
Remove usage of utils.sh

jburel committed:
Commit so travis build with latest

The changes required were not included in 5.3.0-m9 also
latest/omero will point to 5.3.0 as soon as it is released

jburel committed:
Install libjpeg-devel

joshmoore commented:
re: c93e05afd7e7ba51962e4c97a340526cf74524b8 -- I'm not sure how it is to keep the `5.3`-specific parts while dropping the `5.2`- and `5.1`-specific parts, but having this repo "know" that it works with 5.3 could still be a good thing, so that when `/latest/omero/` moves to 5.4 (or later) this repo will keep working. Then we can explicitly update this repo to support 5.4 or choose not to so that it keeps being 5.3 only.

sbesson commented:
Although it might mean regular review, I agree with https://github.com/ome/omero-install/pull/146#issuecomment-288028623 especially since the content of this repository is directly exposed in the public documentation for  OMERO 5.3.x.
More generally, I think that as long as in our OMERO versioning policy remains to allow breaking requirements changes as part of a minor version increment, we should stay on the safe side and favor the usage of minor series redirects i.e. `latest/omeroX.Y`.

dominikl commented:
Tried a docker build of `ubuntu1604_nginx`, got a:
```
+ file=/home/omero/OMERO.server/share/web/requirements-py27-all.txt
+ pip install -r /home/omero/OMERO.server/share/web/requirements-py27-all.txt
Could not open requirements file: [Errno 2] No such file or directory: '/home/omero/OMERO.server/share/web/requirements-py27-all.txt'
```

jburel commented:
@dominikl: This is b/c you did not specify the omero version i.e. it will have been built with the latest official release 5.2.7

dominikl commented:
Ah, of course, thanks.

jburel committed:
Use specific version

jburel commented:
@joshmoore @sbesson added 5.3 so it is explicit
same will need to be done in https://github.com/ome/omeroweb-install/pull/12
cc @aleksandra-tarkowska

jburel committed:
Do not use requirements-all

jburel committed:
Add comment to be used in doc

dominikl commented:
Tried a couple of docker builds yesterday, worked fine. Will try some again today, with the latest commits.

dominikl commented:
Tried a `ubuntu1604_nginx` and a`centos6_py27_nginx` docker build, no problems 👍

mtbc commented on linux/step05_ubuntu1404_nginx.sh:
```diff
@@ -2,14 +2,10 @@
-p=nginx
-
 #start-install
 # require to install more recent version of nginx
```
maybe "required"

mtbc commented on linux/step05_ubuntu1404_nginx.sh:
```diff
@@ -2,14 +2,10 @@
-
 #start-install
 # require to install more recent version of nginx
 # w/o the version installed is 1.4.6
```
maybe expand "w/o" to "without this,"

jburel committed:
Fix typo

jburel commented:
The change in ``travis.yml`` will be reverted before tagging
